### PR TITLE
ROX-13031: DryRun Collection API

### PR DESCRIPTION
**Pull request by keyallis**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Review by charmik-redhat — COMMENTED** [review]
Left a comment on changing error message, looks good otherwise. 

Also, I think you can add a unit test that checks that correct dryRun func on datastore is invoked based on whether id is set or not.

File: central/resourcecollection/service/service_impl.go [hunk 1]
```diff
@@ -83,7 +83,17 @@ func (s *serviceImpl) GetCollection(ctx context.Context, request *v1.GetCollecti
 	if request.GetId() == "" {
 		return nil, errors.Wrap(errox.InvalidArgs, "Id field should be set when requesting a collection")
```
Comment (md2119) [inline central/resourcecollection/service/service_impl.go:84]: nit `'id' field` or `Id should be set`

File: central/resourcecollection/service/service_impl.go [hunk 2]
```diff
@@ -240,3 +250,31 @@ func (s *serviceImpl) ListCollections(ctx context.Context, request *v1.ListColle
 		Collections: collections,
 	}, nil
 }
+
+func (s *serviceImpl) DryRunCollection(ctx context.Context, request *v1.DryRunCollectionRequest) (*v1.DryRunCollectionResponse, error) {
+	if !features.ObjectCollections.Enabled() {
+		return nil, errors.New("Resource collections is not enabled")
```
[the remark this answers is not in the document]
Comment (md2119) [reply central/resourcecollection/service/service_impl.go:256]: +1

**Review by md2119 — CHANGES_REQUESTED** [review]
minor nits otherwise lgtm

**Review by md2119 — APPROVED** [review]

**Pull request merged**